### ISSUE TITLE
DNS manager not accessible when Jetpack is broken on an Atomic site

### ISSUE THREAD

**Issue by arunsathiya** [Automattic/wp-calypso#48746]
#### Steps to reproduce

1. Starting at URL: Start with an Atomic website (a WordPress.com website with a plugin or third-party theme installed). 
2. Break its Jetpack connection (unclear how to do that though)
3. When it's broken, try visiting the domain's DNS manager on Calypso.

#### What I expected

I am able to manage DNS for domain connected to an Atomic site despite having its Jetpack connection broken.

#### What happened instead

I see a blank, white screen on Calypso instead, with no ability to manage DNS for the domain.

#### Browser / OS version

Not applicable, applies to all browsers and OS.

#### Screenshot / Video

<img width="1792" alt="Screenshot on 2021-01-09 at 09-18-01" src="https://user-images.githubusercontent.com/18581859/104088896-0b1ab500-5290-11eb-975e-1e04c6f0f683.png">

#### Context / Source

This came up during a conversation with a customer. 26879468-hc.

#### Notes

As of now, this involves a Happiness Engineer (customer support staff at Automattic) to change the primary domain of the site to `.wpcomstaging.com` address, reconnect Jetpack and switch back to the other custom domain. By reconnecting Jetpack in this manner, DNS zone is visible again on Calypso. 

I feel that it's worthwhile to explore an improvement to this flow. 

cc @Automattic/cobalt as you are more in tune and might know if this bug report belongs on Calypso or internally.

**Comment by klimeryk** [comment]
@aidvu or someone else from Atomic - could look into this? :pray: Cobalt is happy to help but this heavily touches Atomic _and_ Jetpack and my guess is that you have the most experience and context around it 🙇‍♂️

**Comment by aidvu** [comment]
This has been a long standing issue. We're trying to make Jetpack connections unbreakable. W/o that, I think it's not going to be easy to fix this. It would take quite a while to untangle all the stuff. /cc @brandonpayton

**Comment by katiebethbrown** [comment]
> Break its Jetpack connection (unclear how to do that though)

One cause of this is when the primary domain is pointed away from WordPress.com. That breaks the Jetpack connection and makes all Calypso settings inaccessible. 

We generally resolve this by making the staging domain primary again, which allows us to reconnect so that the settings are accessible.

Could we approach this by detecting when the primary domain is pointed away and automatically falling back to the staging domain in that case?

**Comment by aidvu** [comment]
> Could we approach this by detecting when the primary domain is pointed away and automatically falling back to the staging domain in that case?

This is kinda hard, but doable. I'll let @klimeryk chime in if we have anything currently for similar situations.

Another way would be to stop tying specific upgrades like domains to a site. :) Did we ever go ahead and make that single domains page Igor?

**Comment by klimeryk** [comment]
> Another way would be to stop tying specific upgrades like domains to a site. :) Did we ever go ahead and make that single domains page Igor?

@aidvu, yes and no: https://wordpress.com/domains/manage
It's still being pulled from mapping records, AFAIK. So needs to be tied to a site. And there could be some checks on the backed to filter out Jetpack sites. But I'm sure that can be tweaked as needed. Let me know if you want to pair on this, but the code is in the usual place :)

> Could we approach this by detecting when the primary domain is pointed away and automatically falling back to the staging domain in that case?

Hmmm, I'm afraid not 🤔 But! We could add a Domain Medic checker and fixer for this situation. But DM runs on each domain once a month, so there could be a delay before the site is automatically fixed :/ And I'm worried that automatically fixing these situations could not always be "appreciated" - for example, there could be a temporary DNS issue. And we would switch the domain to the staging subdomain and the user would be confused. We can overcome that by sending an email to the site owner (or all admins on site?).

But if possible, I'd love to address the root issue here first - which is how "brittle" the Jetpack connection seems to be. If we could gracefully handle Jetpack connection issues like this - but I have no idea of the technical reasons for that and I'm sure if it was easy to overcome we'd have done it already 😬

**Comment by aidvu** [comment]
> It's still being pulled from mapping records, AFAIK. So needs to be tied to a site. And there could be some checks on the backed to filter out Jetpack sites. But I'm sure that can be tweaked as needed. Let me know if you want to pair on this, but the code is in the usual place :)

This might be a nice idea to unblock HEs for such sites. Can do some pairing this week!

> Hmmm, I'm afraid not thinking But! We could add a Domain Medic checker and fixer for this situation. But DM runs on each domain once a month, so there could be a delay before the site is automatically fixed :/ And I'm worried that automatically fixing these situations could not always be "appreciated" - for example, there could be a temporary DNS issue. And we would switch the domain to the staging subdomain and the user would be confused. We can overcome that by sending an email to the site owner (or all admins on site?).

I was also thinking whether this is replacing one problem for another. I think either a "quick" fix to show all domains regardless of the Jetpack connection state, or fixing the connections somehow is better.

**Comment by klimeryk** [comment]
Yup, definitely agree that fixing the connections would be the best solution here. If that's not possible short-term, we can look into showing all domains regardless of the Jetpack connection state. Happy to pair on it!

**Comment by katiebethbrown** [comment]
Another report: 28246365-hc

User has a domain registered here, pointing to another host via Cloudflare. I switched primary back to staging domain and reconnected JP so they could manage DNS.

**Comment by katiebethbrown** [comment]
Another report: 3820032-zd-woothemes

Less specific to domains though. More concerned with a renewal that happened that they can't see to manage since their primary domain is pointing to Wix and JP is broken.

**Comment by zdenys** [comment]
3872476-zd-woothemes

User has a self-hosted JP connected site and cannot manage the domain settings due to the error in Calypso.

**Comment by katiebethbrown** [comment]
Another report in 3908352-zd-woothemes

User transferred in a custom domain. It has our nameservers but still has an A record pointing away. When it got set as primary after the transfer that broke JP so they couldn't correct the DNS issue without our intervention (I set the staging domain as primary again).

**Comment by katiebethbrown** [comment]
Another report in 28899496-hc

User mapped a custom domain and set it as primary before they configured DNS, which broke JP and the site was no longer able to be managed or worked on in any capacity.

I set the staging domain as primary again to resolve while they work on updating the DNS for the mapped domain.

**Comment by katiebethbrown** [comment]
Another report in 3919112-zd-woothemes

User mapped domain and set it as primary before configuring DNs, which broke JP and the site was no longer able to be managed or worked on in any capacity.

I set the staging domain as primary again to resolve while they work on updating the DNS for the mapped domain.

**Comment by katiebethbrown** [comment]
Another report in 29271331-hc

User has primary domain pointed to another host with an A record. Setting the primary back to the staging domain reveals that they've also made database changes to the site here and possibly dropped required tables, so we aren't able to connect Jetpack to the staging domain either. This leaves their purchased domain and settings completely inaccessible.

**Comment by katiebethbrown** [comment]
Another report in 1266520-hc / p9F6qB-6SP-p2

User migrated a site in, which broke Jetpack, so they weren't able to access their domain settings (or anything in Calypso at all).

I fixed the JP connection and they are back up and going.

**Comment by katiebethbrown** [comment]
Another report in 3975909-zd-woothemes

User mapped a domain, and then followed our steps to update DNS to point to the site here. They were surprised when the old site and their email stopped loading, and changed the DNS to point back to it, which broke Jetpack and left them with a completely inaccessible site here with just a JP connection error showing when they login to wpcom.

I set the staging domain as primary, which restored the connection so they can again work on their site.

**Comment by aamills** [comment]
This happened in 23696284-hc and caused a lot of confusion, both for the user and internally.

Site is now reverted and user is able to access the domain and DNS as they expected, but it was a process to get there.

**Comment by katiebethbrown** [comment]
Another report here: 17019222-hc

Set the staging domain as primary for the user so they could access the DNS setting for their domain that's now pointed elsewhere.

**Comment by arunsathiya** [comment]
Another issue when Jetpack is broken is, purchases like domain registrations and email hosting (Professional Email or Google Workspace) are not accessible too. Visiting https://wordpress.com/me/purchases shows that the user does not have permissions to manage them.

![image](https://user-images.githubusercontent.com/18581859/134806560-cfc92a80-78e8-4c96-9853-3bff19dba9e5.png)

**Comment by Sandkorner** [comment]
Just wanted to add to the above comment, that it helped to change to the wpcomstaging domain and deactivate all plugins, also not go straight to the manage purchases options, but to go to my account and then there. I removed the professional email for the user.

**Comment by zdenys** [comment]
4922231-zd-woothemes

**Comment by kriskorn** [comment]
This issue happens only in specific instances now where Jetpack is disconnected via Jetpack Debugger `Disconnect Jetpack` link. In any other case, Calypso is accessible and DNS can be managed as expected, even if sites have fatal errors or the domain is pointed to an external provider. Marking this as solved for now. 

Context: pdtR3Z-1Cq-p2#comment-1504 & p1679414671950499-dotcom-reactor
